controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -2088,3 +2089,88 @@ func changeClusterNameFails(t *testing.T, ctx context.Context, f *framework.Fram
 	require.NoError(err, "failed to delete K8ssandraCluster")
 	f.AssertObjectDoesNotExist(ctx, t, dc1Key, &cassdcapi.CassandraDatacenter{}, timeout, interval)
 }
+
+// Create a cluster with server and client encryption but client encryption stores missing.
+// Verify that dc1 never gets created.
+func injectContainers(t *testing.T, ctx context.Context, f *framework.Framework, namespace string) {
+	require := require.New(t)
```

Comment:
This test should probably have Medusa enabled and then also verify positions of medusa-restore and jmx-credentials init containers in addition to user-defined ones.